VMBackup/main/safefreeze/src/safefreeze.c

```diff
@@ -103,15 +105,15 @@ int main(int argc, char *argv[])
 
         if ((fileSystemDescriptors[i] = open(mountPoint, O_RDONLY)) < 0)
         {
-            logger("Failed to open: %s\n", mountPoint);
+            logger("Failed to open: %s with error: %d and error message: %s\n", mountPoint, fileSystemDescriptors[i], strerror(errno));
```

Comment:
BUG. errno is a global variable that must be captured `*immediately*` after the failing system call/library call. Subsequent calls like printf/time/etc. that you are calling in logger will overwrite it. Please fix it in all places below.

See the NOTES section in the `man errno` page. Quoting here some parts

```
NOTES
       A common mistake is to do

           if (somecall() == -1) {
               printf("somecall() failed\n");
               if (errno == ...) { ... }
           }

       where  errno  no  longer  needs  to  have the value it had upon return from somecall() (i.e., it may have been
       changed by the printf(3)).  If the value of errno should be preserved across a library call, it must be saved:

           if (somecall() == -1) {
               int errsv = errno;
               printf("somecall() failed\n");
               if (errsv == ...) { ... }
           }
```